dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/treemap_config_panel_item.tsx

```diff
@@ -110,7 +124,63 @@ export const TreemapConfigPanelItem = ({
       : unselectedFields;
   };
 
-  return (
+  const options = getOptionsAvailable(GROUPBY).map((opt) => ({
+    label: opt.label,
+    name: opt.name,
+  }));
+
+  const renderParentPanel = () => {
+    const selectedAxis = configList.dimensions[0]?.parentFields;
+    const { index } = selectedParentItem;
+    return (
+      <>
+        <DataConfigItemClickPanel
+          isSecondary
+          title={`Parent ${index + 1}`}
+          closeMenu={() => isHandlePanelClickBack(selectedAxis)}
+        />
+        <EuiComboBox
+          id={htmlIdGenerator('axis-select-')}
+          placeholder="Select a field"
+          options={options}
+          selectedOptions={selectedAxis[index].label !== '' ? [selectedAxis[index]] : []}
+          isClearable={true}
+          singleSelection={{ asPlainText: true }}
+          onChange={handleParentChange}
+          aria-label="Parent field"
+        />
+      </>
+    );
+  };
+  const handleUpdateParentFields = (arr: ParentUnitType[]) =>
+    updateList(GROUPBY, PARENTFIELDS, arr);
+
+  const handleParentChange = (values: Array<EuiComboBoxOptionOption<unknown>>) => {
```

Comment:
Please add comments as much as we can for all these functions.

Reply:
Done

Reply:
Added block comments in functions